js/misc/paygManager.js

```diff
@@ -86,18 +88,23 @@ var PaygManager = new Lang.Class({
     },
 
     _onProxyConstructed: function(object, res) {
+        let success = false;
         try {
-            object.init_finish (res);
+            success = object.init_finish (res);
         } catch (e) {
             logError(e, "Error while constructing D-Bus proxy for " + EOS_PAYG_NAME);
-            return;
         }
 
-        this._setEnabled(this._proxy.Enabled);
-        this._setExpiryTime(this._proxy.ExpiryTime);
+        if (success) {
+            this._enabled = this._proxy.Enabled;
+            this._expiryTime = this._proxy.ExpiryTime;
```

Comment:
Why not call `this._setEnabled()` and `this._setExpiryTime()`? Are you deliberately wanting to avoid emitting their signals before `initialized` is emitted?

Reply:
YES, otherwise that confuses the shell on the way the screenshield handles the initial state, so I'd rather avoid that emission if you don't mind.

Reply:
Add a comment in the code about that, and then I think we’re good to merge.